pkgs/development/libraries/libid3tag/default.nix

```diff
@@ -12,7 +12,11 @@ stdenv.mkDerivation rec {
   outputs = [ "out" "dev" ];
   setOutputFlags = false;
 
-  propagatedBuildInputs = [ zlib gperf ];
+  strictDeps = true;
+
+  nativeBuildInputs = [ gperf_3_0 ];
+
+  buildInputs = [ zlib ];
```

Comment:
IIUC this should be ok because zlib is in the RUNPATH

```
RUNPATH              /nix/store/dd1icwvgpp5ayh8gzpr1mjhwrrww24xc-zlib-aarch64-unknown-linux-gnu-1.2.12/lib:/nix/store/fkjwkg63nkg1xylj9syzqwlh441z8vms-glibc-aarch64-unknown-linux-gnu-2.35-163/lib
```